Un-OOP the tutorial

NogginBops opened the pull request:
This PR removes a lot of the unnecessary object oriented abstractions the tutorial has make makes it clearer for an OpenGL beginner what is actually going on.

One thing I've not yet done with this implementation is removed the "Common" files entirely, but that is basically the goal with this PR. Just make the code easier to read and less obfuscated by placing all the code for each chapter in it's own project so that it's easier to see what code is actually being used.

Fixes #93
Fixes #95
Fixes #96
Fixes #48

NogginBops committed:
un-oop:ed the Shader class

NogginBops committed:
Un-oop:ed the Shader and Texture classes some more.

MauNguyenVan commented:
Actually, this PR helps learners see the true nature of OpenGL instead of the abstraction.  The code will be easier to read and clearer to understand OpenGL

NogginBops committed:
Removed the Common project in favor of keeping the code in each chapter standalone.
Removed the shader class in favour of the more straight forward int way.
Changed to use ClientSize instead of Size in NativeWindowSettings.
Updated to OpenTK 4.9.4.
Updated StbImageSharp to 2.30.15.
Revised a few comments.
Removed all OpenTK.dll.config files.

NogginBops committed:
Fix framebuffer resize event.
